### PR TITLE
Fix multinomial logpmf validity mask for batched input

### PR DESCRIPTION
**Pull request by VaggelisGian**
## Description

`jax.scipy.stats.multinomial.logpmf` checked its support condition with `jnp.sum(x)` over all axes instead of the category axis. For batched input the mask therefore compared a single global total against `n`, so a batch containing one row whose counts did not sum to `n` returned `-inf` for every row, and a batch whose grand total coincidentally equaled `n` returned finite garbage for invalid rows (for example `logpmf([[10,0,0],[0,0,0]], 10, p)` returned about `+15.1` for the impossible zero row). The fix sums over the last axis only, matching the per-row semantics of `scipy.stats.multinomial`.

Found by differential testing against scipy 1.17.1 while auditing `jax.scipy.stats` boundary behavior; the existing parametrized test always generated rows summing to `n`, so the bug was invisible to it. The new regression test covers a fully valid batch, a wrong-total row, and the grand-total coincidence case.

## AI assistance disclosure

This change was developed with AI assistance under my direction and review; I have examined the code and tests, run the verification below locally, and submit this in accordance with the project's AI policy and the Google CLA.

## Test plan (run locally on Windows, Python 3.12, scipy 1.17.1)

```
python -m pytest tests/scipy_stats_test.py -k Multinomial -q
  4 passed, 979 deselected
JAX_ENABLE_X64=1 python -m pytest tests/scipy_stats_test.py -k Multinomial -q
  11 passed, 1011 deselected
python -m pytest tests/scipy_stats_test.py -q
  973 passed, 10 skipped
python -m ruff check jax/_src/scipy/stats/multinomial.py tests/scipy_stats_test.py
  All checks passed!
```

Before/after on the regression case:

```
before: logpmf([[10,0,0],[4,3,3]], 10, p) = [-inf, -inf]
after : logpmf([[10,0,0],[4,3,3]], 10, p) = [-16.09438, -3.78627]   (scipy agrees)
before: logpmf([[10,0,0],[0,0,0]], 10, p) = [-16.09438, +15.10441]
after : logpmf([[10,0,0],[0,0,0]], 10, p) = [-16.09438, -inf]       (scipy agrees)
```

### CODE REVIEW

**Review by CAOShurong — COMMENTED** [review]
Verified end-to-end on Windows (CPython 3.13.1, CPU jaxlib) against the exact PR head `18f025a7afb3fa5c4388597f3a74a8121691ad2f` and current main (`8c191b842df0`). This fix is correct and I can confirm both failure modes independently:

**Pre-fix reproduction (main `8c191b84`)**

1. *Batch poisoning*: `logpmf([[10,0,0],[4,3,3]], 10, p)` — every row valid — returned `[-inf, -inf]` because `jnp.sum(x)` reduces over all axes, so the mask compared the global total (20) against `n=10`. Any batch containing one row whose counts don't sum to `n` gets `-inf` everywhere.
2. *Grand-total coincidence*: `logpmf([[10,0,0],[0,0,0]], 10, p)` returned `[-16.094, +15.104]` — the impossible zero row got finite garbage (+15.1) because the global total coincidentally equaled `n`, while scipy correctly returns `-inf` for that row.

**Post-fix verification (PR head `18f025a7`)**

- Case 1 now returns `[-16.09438, -3.78627]`, matching scipy to float32 precision; the invalid-row case correctly returns `-inf` for that row only.
- Case 2 now matches scipy (`-inf` for the zero row, `-16.09438` for the valid row).
- Bonus check beyond the PR's tests: the new `jnp.equal(jnp.sum(x, axis=-1), n)` also broadcasts correctly when `n` itself is batched per-row (e.g. `x` shape `(3,3)`, `n = [10,10,9]`) — per-row validity with mixed `n` values matches scipy. Worth noting this works because `jnp.sum(x, axis=-1)` yields shape `x.shape[:-1]`, which broadcasts elementwise against `n`; no further change needed.
- Full `tests/scipy_stats_test.py` at the PR head: **973 passed, 10 skipped** (matches the author's claimed numbers). `ruff check` clean on the two changed Python files.
- The CHANGELOG entry under a "Bug fixes" heading follows the repo convention (most fixes here land without one, but it's accurate and well-written).

One observation, not blocking: the existing parametrized `_CheckAgainstNumpy` cases never caught this because their generated rows always sum to `n`. The three added regression batches cover exactly the interesting partition (valid batch / wrong-total row / grand-total coincidence), which is the right minimal set.

LGTM from my side — nice find, and the root-cause explanation (axis-less reduction making the support check a global predicate instead of a per-row predicate) matches the code history: the axis argument was lost when this was written rather than regressed later.

**Comment by VaggelisGian** [comment]
Friendly ping: the Google CLA was signed yesterday - cla/google already passes on my two newest PRs (#40227, #40228) - but this PR's check still shows the failure from before signing. Not urgent; just flagging in case the check needs a fresh run to pick up the signature.
